lib/libvfio-user.c

```diff
@@ -862,17 +852,13 @@ MOCK_DEFINE(exec_command)(vfu_ctx_t *vfu_ctx, struct vfio_user_header *hdr,
             *iovecs = _iovecs;
             *nr_iovecs = 2;
         } else {
-            ret = -errno;
             free(irq_info);
```

Comment:
Won't `free` overwrite `errno` set by `handle_device_get_irq_info`?

Reply:
You're technically right. I always forget that this isn't guaranteed (yet) by Linux.

https://sourceware.org/bugzilla/show_bug.cgi?id=17924

Surprised that is so recently fixed.